ui/arduino/store.js

```diff
@@ -166,7 +166,7 @@ function store(state, emitter) {
     let editor = state.cache(AceEditor, 'editor').editor
     let contents = cleanCharacters(editor.getValue())
     editor.setValue(contents)
-    let filename = state.selectedFile || 'undefined'
+    let filename = cleanCharacters(state.selectedFile) || 'undefined'
```

Comment:
Shouldn't it be `let filename = cleanCharacters(state.selectedFile || 'undefined')`?

Reply:
I think `'undefined'` is a string value given arbitrarily, and it will never contain non ASCII characters, while the filename given by the user might, especially on keyboards which have non-ASCII characters directly available from the keyboard without modifier keys

Reply:
I question this bogus case when `state.selectedFile` is `null`.

If you call `cleanCharacters(null)` it will fail. So the correct code is either:


```ts
let filename = state.selectedFile ? cleanCharacters(state.selectedFile) || 'undefined'
```

or this:

```ts
let filename = cleanCharacters(state.selectedFile || 'undefined')
```

Reply:
oh, I think I got it now
so if it's `null` it will take `'undefined'` as input?

Reply:
Yes, and the ASCII-conversion will run, but won't change anything.

Reply:
Duplicate of https://github.com/arduino/lab-micropython-editor/pull/58/files#r1255709002